lib/validate.js

```diff
@@ -8,7 +8,7 @@ const _ = require('lodash');
 
 function getEntryForFunction(serverlessFunction) {
 	const handler = serverlessFunction.handler;
-	const handlerFile = /(.*)\..*?$/.exec(handler)[1] + '.js';
+	const handlerFile = /(.*)\..*?$/.exec(handler)[1];
```

Comment:
We should actually retrieve the correct file extension here and set it in the value of the key, so that there's no need to use custom code in the webpack configuration to get the plugin running.

```
function getHandlerFileExtension(baseName) {
  // Use fs/fs-extra/path or any other appropriate mechanism to determine the actual file extension of the handler
  // Maybe we can search for known and supported endings in a predefined order
  // So that the standard extension 'js' is preferred if, by accident, there are multiple copies
  // of the handler with different extensions - or throw an error in that case.
  const allowedExtensions = [
    'js', 'ts', 'jsx'
  ];
  const handlerExtension = ...;
  ...
  return handlerExtension;
}
...
const handlerFileBaseName = /(.*)\..*?$/.exec(handler)[1];
const handerFile = `${handlerFileBaseName}.${getHandlerFileExtension(handlerFileBaseName)}`;
...
return {
  [handlerFileBaseName]: `./${handlerFile}`
};
```